integration/simulation/utils.go

```diff
@@ -56,29 +58,22 @@ func createInMemObscuroNode(id int64, genesis bool, avgGossipPeriod uint64, avgB
 	enclaveClient := enclave.NewEnclave(nodeID, true, stats)
 
 	// create an in memory obscuro node
-	node := host.NewObscuroAggregator(nodeID, obscuroNodeCfg, nil, stats, genesis, enclaveClient, obscuroInMemNetwork)
+	node := host.NewObscuroAggregator(nodeID, obscuroNodeCfg, nil, stats, genesis, enclaveClient, obscuroInMemNetwork, placeholderAddress)
 	obscuroInMemNetwork.currentNode = &node
 	return &node
 }
 
-func createSocketObscuroNode(id int64, genesis bool, avgGossipPeriod uint64, stats *Stats, peerAddrs []string) *host.Node {
+func createSocketObscuroNode(id int64, genesis bool, avgGossipPeriod uint64, stats *Stats, peerAddrs []string, enclavePort uint64) *host.Node {
 	nodeID := common.BigToAddress(big.NewInt(id))
 
-	// create a remote enclave server
-	enclavePort := uint64(enclaveStartPort + id)
-	err := enclave.StartServer(enclavePort, nodeID, stats)
-	if err != nil {
-		panic(fmt.Sprintf("failed to create enclave server: %v", err))
-	}
-
 	// create an enclave client
 	enclaveAddr := fmt.Sprintf("%s:%d", localhost, enclavePort)
-	enclaveClient := host.NewEnclaveRPCClient(enclaveAddr, host.ClientRPCTimeoutSecs*time.Second)
+	enclaveClient := host.NewEnclaveRPCClient(enclaveAddr, host.ClientRPCTimeoutSecs*time.Second, nodeID)
 
 	// create a socket obscuro node
 	nodeP2p := p2p.NewSocketP2PLayer(peerAddrs[id-1], peerAddrs)
```

Comment:
this peerAddrs[id-1] is a bit unclear.
I would pass in the "ourAddress" from the caller.
The ids will not be consecutive numbers forever.

Reply:
Will fix, good point.